frontend/packages/console-app/src/components/nodes/NodeTerminal.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import { Firehose, FirehoseResult, LoadingBox } from '@console/internal/components/utils';
+import { NodeKind, PodKind } from '@console/internal/module/k8s';
+import { PodExecLoader } from '../../../../../public/components/pod';
+import { PodModel } from '../../../../../public/models';
+import { k8sCreate, k8sKill } from '../../../../../public/module/k8s';
+
+type NodeTerminalInnerProps = {
+  obj?: FirehoseResult<PodKind>;
+};
+
+type NodeTerminalProps = {
+  obj: NodeKind;
+};
+
+const getDebugPod = (node) => {
+  return {
+    kind: 'Pod',
+    apiVersion: 'v1',
+    metadata: {
+      generateName: `${node.metadata.name}-debug-`,
+      namespace: 'default',
+      creationTimestamp: null,
+      annotations: {
+        'debug.openshift.io/source-container': 'container-00',
+        'debug.openshift.io/source-resource': `/v1, Resource=nodes/${node.metadata.name}`,
+      },
+    },
+    spec: {
+      volumes: [
+        {
+          name: 'host',
+          hostPath: {
+            path: '/',
+            type: 'Directory',
+          },
+        },
+      ],
+      containers: [
+        {
+          name: 'container-00',
+          image: 'registry.redhat.io/rhel7/support-tools',
+          command: ['/bin/sh'],
+          resources: {},
+          volumeMounts: [
+            {
+              name: 'host',
+              mountPath: '/host',
+            },
+          ],
+          securityContext: {
+            privileged: true,
+            runAsUser: 0,
+          },
+          stdin: true,
+          stdinOnce: true,
+          tty: true,
+        },
+      ],
+      restartPolicy: 'Never',
+      nodeName: `${node.metadata.name}`,
+      hostNetwork: true,
+      hostPID: true,
+    },
+    status: {},
+  };
+};
+
+const podIsRunning = (obj) => {
+  return obj.data.status.phase === 'Running';
+};
+
+const NodeTerminalInner: React.FC<NodeTerminalInnerProps> = ({ obj }) => {
+  if (!obj || !obj.loaded || !podIsRunning(obj)) {
+    return <LoadingBox />;
+  }
+  return <PodExecLoader obj={obj.data} isNode />;
+};
+
+const NodeTerminal: React.FC<NodeTerminalProps> = ({ obj: node }) => {
+  const [resources, setResources] = React.useState([]);
+  React.useEffect(() => {
+    const debugPod = getDebugPod(node);
+    k8sCreate(PodModel, debugPod)
+      .then((pod) => {
+        setResources([
+          {
+            isList: false,
+            kind: 'Pod',
+            name: pod.metadata.name,
+            namespace: pod.metadata.namespace,
+            prop: 'obj',
+          },
+        ]);
+      })
+      // eslint-disable-next-line no-console
+      .catch((e) => console.log(e));
+    return () => k8sKill(PodModel, debugPod);
```

Comment:
I believe you that this works, but I'm not sure how :/ `debugPod` shouldn't have a name since we use `generateName`

Reply:
We might try to delete the pod on a window `unload` event as well in case the user closes the browser tab while on this page.

https://developer.mozilla.org/en-US/docs/Web/API/Window/unload_event

Reply:
We should also avoid trying to delete the pod if creation failed.